nsot/models.py

```diff
@@ -1470,6 +1470,93 @@ def to_dict(self):
         }
 
 
+class Circuit(Resource):
+    """Represents two network Interfaces that are connected"""
+
+    # a_endpoint interface
+    a_endpoint = models.ForeignKey(
+        Interface, db_index=True, related_name='circuits_a',
+        null=False, verbose_name='A-side Interface',
+        help_text='Unique ID of interface at the A-side'
+    )
+
+    # z_endpoint interface
+    z_endpoint = models.ForeignKey(
+        Interface, db_index=True, related_name='circuits_z',
+        null=True, verbose_name='Z-side Interface',
+        help_text='Unique ID of interface at the Z-side'
+    )
+
+    # We are currently inferring the site_id from the parent (A-side) Interface
+    # in the .save() method
+    site = models.ForeignKey(
+        Site, db_index=True, related_name='circuits',
+        on_delete=models.PROTECT,
+        help_text='Unique ID of the Site this Circuit is under'
+    )
+
+    display_name = models.CharField(
+        max_length=255, unique=True, help_text='Unique display name of circuit'
+    )
+
+    def __unicode__(self):
+        return u'%s' % self.display_name
+
+    class Meta:
+        unique_together = ('a_endpoint', 'z_endpoint')
+        index_together = unique_together
+
+    def clean_site(self, value):
+        """Always enforce that site is set."""
+        if value is None:
+            try:
+                return self.a_endpoint.device.site_id
+            except Device.DoesNotExist:
+                return Device.objects.get(id=self.a_endpoint.device_id).site_id
+
+        return value
+
+    def clean_display_name(self, value, delimeter='--'):
+        """Always enforce that display_name is set"""
+        # Add display name of hostname:intf::hostname:intf
+        # '--' as default delimeter because an intf name may contain '-'
+        if value is None:
+            a_display_name = '{}:{}'.format(
+                self.a_endpoint.device.hostname,
+                self.a_endpoint.name,
+            )
+            if getattr(self.z_endpoint.name):
```

Comment:
There's a `TypeError` here if `self.display_name=None` because the field is currently named `display_name` (not `name`).